asserts/account_key_test.go

```diff
@@ -0,0 +1,150 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts_test
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/ubuntu-core/snappy/asserts"
+)
+
+type accountKeySuite struct {
+	pubKeyBody           string
+	fp                   string
+	since, until         time.Time
+	sinceLine, untilLine string
+}
+
+var _ = Suite(&accountKeySuite{})
+
+func (aks *accountKeySuite) SetUpSuite(c *C) {
+	pk, err := asserts.GeneratePrivateKeyInTest()
+	c.Assert(err, IsNil)
+	aks.fp = hex.EncodeToString(pk.PublicKey.Fingerprint[:])
+	aks.since, err = time.Parse(time.RFC822, "16 Nov 15 15:04 UTC")
+	c.Assert(err, IsNil)
+	aks.until = aks.since.AddDate(1, 0, 0)
+	buf := new(bytes.Buffer)
+	err = pk.PublicKey.Serialize(buf)
+	c.Assert(err, IsNil)
+	aks.pubKeyBody = "openpgp " + base64.StdEncoding.EncodeToString(buf.Bytes())
+	aks.sinceLine = "since: " + aks.since.Format(time.RFC3339) + "\n"
+	aks.untilLine = "until: " + aks.until.Format(time.RFC3339) + "\n"
+}
+
+func (aks *accountKeySuite) TestDecodeOK(c *C) {
+	encoded := "type: account-key\n" +
+		"authority-id: canonical\n" +
+		"account-id: acc-id1\n" +
+		"fingerprint: " + aks.fp + "\n" +
+		aks.sinceLine +
+		aks.untilLine +
+		fmt.Sprintf("body-length: %v", len(aks.pubKeyBody)) + "\n\n" +
+		aks.pubKeyBody + "\n\n" +
+		"openpgp c2ln"
+	a, err := asserts.Decode([]byte(encoded))
+	c.Assert(err, IsNil)
+	c.Check(a.Type(), Equals, asserts.AccountKeyType)
+	accKey := a.(*asserts.AccountKey)
+	c.Check(accKey.AccountID(), Equals, "acc-id1")
+	c.Check(accKey.Since(), Equals, aks.since)
+	c.Check(accKey.Until(), Equals, aks.until)
+}
+
+const (
+	errPrefix = "assertion account-key: "
+)
+
+func (aks *accountKeySuite) TestDecodeInvalidHeaders(c *C) {
+	encoded := "type: account-key\n" +
+		"authority-id: canonical\n" +
+		"account-id: acc-id1\n" +
+		"fingerprint: " + aks.fp + "\n" +
+		aks.sinceLine +
+		aks.untilLine +
+		fmt.Sprintf("body-length: %v", len(aks.pubKeyBody)) + "\n\n" +
+		aks.pubKeyBody + "\n\n" +
+		"openpgp c2ln"
+
+	for _, scen := range []struct {
+		original, invalid, expectedErr string
+	}{
+		{"account-id: acc-id1\n", "", "account-id header is mandatory"},
```

Comment:
This will probably look better as a top-level variable, along the lines of:
```
var invalidHeadersTests = []struct{ original, invalid, expectedErr string }{
        ...
}
...
        for _, test := range invalidHeaderTests {
                ...
        }
```